lib/utils/FileCache.mjs

```diff
@@ -34,7 +34,16 @@ export default class FileCache {
 		const useCheckSum = cacheStrategy === CACHE_STRATEGY_CONTENT;
 
 		debug(`Cache file is created at ${cacheFile}`);
-		this._fileCache = fileEntryCache.createFromFile(cacheFile, useCheckSum, undefined);
+
+		try {
+			this._fileCache = fileEntryCache.createFromFile(cacheFile, useCheckSum, undefined);
+		} catch {
+			debug(`Cache file might be corrupt, attempting to remove and recreate the cache file`);
+
+			rmSync(cacheFile, { force: true });
```

Comment:
note

The `recursive` option is false by default and _if_ `cacheLocation` is a directory, Stylelint creates a cache file inside the specified folder.